Revert "Revert "Mk nikshay audit cookie safe""

mkangia opened the pull request:
Reverts dimagi/commcare-hq#16005

This fixes the issue introduced due to changes in app preview.
Adding hold off on merge to add a notification in case csrftoken is missed anywhere to better tracking.

mkangia committed:
Revert "Revert "Mk nikshay audit cookie safe""

dimagimon commented on corehq/apps/hqwebapp/static/hqwebapp/js/ajax_csrf_setup.js:
```diff
@@ -7,7 +7,7 @@ $.ajaxSetup({
         // Don't pass csrftoken cross domain
         if (!csrfSafeMethod(settings.type) && !this.crossDomain) {
-            $csrf_token = $.cookie('csrftoken');
+            $csrf_token = $("#csrfTokenContainer").val();
```
'$csrf_token' is not defined. (no-undef)

mkangia committed:
Add csrftoken block to app_preview base template

Since app_preview runs in an iframe it cant access the DOM element added on the application for csrftoken

mkangia committed:
Add email notification for csrftoken missing

dimagimon commented on corehq/apps/hqwebapp/middleware.py:
```diff
@@ -0,0 +1,24 @@
+        _assert = soft_assert('{}@{}'.format('mkangia+logs', 'dimagi.com'), exponential_backoff=False)
+        _assert(False, warning)
+
+        return super(HQCsrfViewMiddleWare, self)._reject(request, reason)
```
W292 no newline at end of file

mkangia commented:
@sravfeyn thanks for the suggestion for notification. Added!
@orangejenny added csrftoken missed earlier in app preview base view.

orangejenny commented:
Code looks fine to me, but csrf test is failing: `AssertionError: Request at /a/delhi/sms/send_to_recipients/ doesn't contain a csrf token.`

mkangia committed:
fix failing test for csrftoken middleware changes

mkangia commented:
Aha! fixed it. Thanks for flagging @orangejenny

mkangia committed:
fix dimagimon complaint [skip ci]

dimagimon commented on corehq/apps/hqwebapp/middleware.py:
```diff
@@ -0,0 +1,25 @@
+        _assert(False, warning)
+
+        return super(HQCsrfViewMiddleWare, self)._reject(request, reason)
+
```
W391 blank line at end of file

mkangia committed:
fix dimagimon complaint [skip ci]
